File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.c
#include<stdbool.h>

    bool uniformArray(int* nums1, int nums1Size) {
        int min_val=nums1[0];
        bool has_odd=false;

        for (int i=0;i<nums1Size;i++) {
            if (nums1[i]<min_val) {
                min_val=nums1[i];
            }
            if (nums1[i]%2!=0) {
                has_odd=true;
            }
        }
        return (min_val%2!=0)||!has_odd;
    }

File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.java
class Solution {
    public boolean uniformArray(int[] nums1) {
        int minVal=nums1[0];
        boolean hasOdd=false;

        for (int x:nums1) {
            if (x<minVal) {
                minVal=x;
            }
            if (x%2!=0) {
                hasOdd=true;
            }
        }
        return (minVal%2!=0)||!hasOdd;
    }
}